pkg/arvo/app/chat-hook.hoon

```diff
@@ -389,51 +397,80 @@
 |_  bol=bowl:gall
 ++  grp  ~(. grpl bol)
 ::
-++  poke-fix-dms
-  |=  a=%fix-dms
+++  poke-noun
+  |=  a=?(%fix-dm %fix-out-of-sync)
   ^-  (quip card _state)
+  |^
   :_  state
-  %-  zing
-  %+  turn
-    ~(tap by synced)
-  |=  [=path host=ship]
-  ^-  (list card)
-  ?>  ?=([@ @ *] path)
-  =/  =ship  (slav %p i.path)
-  ?:  =(ship our.bol)
-    ::  local dm, no need to do cleanup
-    ~
-  ?:  ?=(^ (groups-of-chat path))
-    ::  correctly initialized, no need to do cleanup
-    ::
-    ~
-  ?.  =((end 3 4 i.t.path) 'dm--')
-    ~
-  :-  =-  [%pass /fixdm %agent [our.bol %chat-view] %poke %chat-view-action -]
-      !>  ^-  action:view
-      [%delete path]
-  =/  new-dm  /(scot %p our.bol)/(crip (weld "dm--" (trip (scot %p ship))))
-  =/  mailbox=(unit mailbox:store)  (chat-scry path)
-  ?~  mailbox
-    ~
-  :~  =-  [%pass /fixdm %agent [our.bol %chat-view] %poke %chat-view-action -]
-      !>  ^-  action:view
-      :*  %create
-          %-  crip
-          (zing [(trip (scot %p our.bol)) " <-> " (trip (scot %p ship)) ~])
-          ''
-          new-dm
-          ship+new-dm
-          [%invite (silt ~[ship])]
-          (silt ~[ship])
-          %.y
-          %.n
-      ==
-    ::
-      =-  [%pass /fixdm %agent [our.bol %chat-store] %poke %chat-action -]
-      !>  ^-  action:store
-      [%messages new-dm envelopes.u.mailbox]
+  ?-  a
+      %fix-dm    (fix-dm %fix-dm)
+      %fix-out-of-sync  (fix-out-of-sync %fix-out-of-sync)
```

Comment:
You could remove the `|=` from the top of these arms and just write `fix-dm` etc here, without calling them with constants.